lit_gpt/lora.py

```diff
@@ -208,8 +210,14 @@ def __init__(
             self.lora_A = nn.Parameter(self.linear.weight.new_zeros((r * sum(enable_lora), in_features)))  # (4, 128)
             enable_q, enable_k, enable_v = enable_lora
             self.kv_embd_size = self.linear.in_features // (n_head // n_query_groups)
-            shape = self.linear.in_features * enable_q + self.kv_embd_size * enable_k + self.kv_embd_size * enable_v
-            self.lora_B = nn.Parameter(self.linear.weight.new_zeros(shape, r))  # (256, 2))
+            # qkv_shapes will be used to split a tensor with weights correctly
+            self.qkv_shapes = tuple(
+                filter(
+                    None,
+                    (self.linear.in_features * enable_q, self.kv_embd_size * enable_k, self.kv_embd_size * enable_v),
+                )
+            )
```

Comment:
I'd find this easier to read

```suggestion
            qkv_shapes = (self.linear.in_features * enable_q, self.kv_embd_size * enable_k, self.kv_embd_size * enable_v)
            self.qkv_shapes = [s for s in qkv_shapes if s is not None]
```